### ISSUE TITLE
example.com and www.example.com site are different

### ISSUE THREAD

**Issue by radhe** [rtCamp/easyengine#443]
If example.com already exist then www.example.com should be allowed to create.

```^_^[root@example.com:/var/www]# ee site create www.example.com --html 
too many values to unpack (expected 2)
O_O[root@example.com:/var/www]# ```

**Comment by rahul286** [comment]
Lets add a new flag to `ee site create` command called `---www`

It will take values
1. auto (default value) - current behavior
2. off - for site only specified domain will be created. This means if `example.com` created no `www.example.com` will be creared. Also for `www.example.com`, no `example.com`.

**Comment by MiteshShah** [comment]
@radhe In most general case we are accessed same site with and without www.

For example: https://www.rtcamp.com and https://rtcamp.com are same site.

Can you give us more details why this is needed. So other users also know why we do this in future.

**Comment by rahul286** [comment]
We have 2 different SSL certificates for www.rtcamp.com and rtcamp.com

So for ease of maintenance and to make it future-proof when EE starts managing SSL certificates, such an option will help.

As of we now we have add both SSL under same folder and manually coded different `server{..}` blocks inside nginx config. 

Also default behavior will stay same. We will be adding a flag to stop this behavior.

**Comment by rahul286** [comment]
If user wants to create both version, he needs to run:

```
ee site create www.example.com --www=off
ee site create example.com --www=off
```

**Comment by jbthechamp** [comment]
so does this mean that two sites need to be managed?  that is quite cumbersome

**Comment by jbthechamp** [comment]
i love your product, it has server pilot beat to the ground on speed as there is no apache, but the non www and www versions is getting me here, i could write a redirect, which is the simplest method, however you know how google feels about redirects :-)

**Comment by bigwilliam** [comment]
@jbthechamp no, the default action is to have 1 domain which handles both `www` and `example.com`, but if you use the --www=off option, then you could literally create two different sites: `example.com` and `www.example.com`

**Comment by jbthechamp** [comment]
Is there a command to enable this new feature for sites already running?  I
tinkered around trying to run site specific command and couldnt seem to
figure it out.  I just updated to new ee version. This is actually awesome
to know.

There should be an - - ssl flag similar to this or an all around command to
enable all versions so one site can be accessed via ssl,  non ssl,  www,
and non www :-).  That would rock for clients that have that type of need
or desire it. .

**Comment by bigwilliam** [comment]
Yes you can do `ee site update example.com --www=off` I assume. But you can also refer to the docs where all the commands are listed: http://docs.rtcamp.com/easyengine/commands/

the --ssl flag is in the works: https://github.com/rtCamp/easyengine/issues/303

**Comment by jbthechamp** [comment]
Ty ill give it a whirl.  I believe thats what I tried late last night but I
was tired and could have mistyped it.  Thanks for the reference I need to
bookmark via delicious as its a reference i frequent.

**Comment by bigwilliam** [comment]
I could be wrong on the `ee site update example.com --www=off` - I have
never needed this feature and don't use it.

But, NGINX is fairly easy to configure, and EE is set to pull any config
files you add to /var/www/example.com/conf/nginx/. So you could always do
this manually for the time being.

**Comment by kymcarlsson** [comment]
@rahul286 How about a more flags and a more modular approach?
ee site create domain.com 
--protocol= < http | https >  #not required (defaults to http)
--port = < 80 | ... >  #not required (defaults to 80 for http and 443 for https)
--subdomain= < www | any-other-subdomain | ... >  #not required
--subdirectory= < subdirectoryname > #not required
--accessible-by= < domain | subdomain | domain-subdomain | basic-auth | proxy > #required if domain.com is specified with a subdomain

**Comment by rahul286** [comment]
Actually www and non-www subdomain case is different.

Though technically www is just a subdomain, people expect it to be mapped to non-www version or other way round.

**Comment by kymcarlsson** [comment]
@rahul286 for mapping see suggested --accessible-by flag in my above

**Comment by rahul286** [comment]
I think we need to handle first:

Map `www.domain.com => domain.com`  When user runs:

```
ee site create domain.com 
```

And map `domain.com => www.domain.com`  When user runs:

```
ee site create www.domain.com 
```

So if you want www visible, specify www in domain name.

In both cases, there will be 2 nginx server block - one for www and one for non-www.

Only case I see when people do something with other block is setting up separate SSL.

e.g. https://www.domain.com/ to https://domain.com/ redirect (or vice-versa)
